frontend/src/metabase/search/components/InfoText.tsx

```diff
@@ -0,0 +1,122 @@
+import { t, jt } from "ttag";
+
+import * as Urls from "metabase/lib/urls";
+
+import { Icon } from "metabase/core/components/Icon";
+import Link from "metabase/core/components/Link";
+
+import Schema from "metabase/entities/schemas";
+import Database from "metabase/entities/databases";
+import Table from "metabase/entities/tables";
+import { PLUGIN_COLLECTIONS } from "metabase/plugins";
+import { getTranslatedEntityName } from "metabase/nav/utils";
+
+import type { Collection } from "metabase-types/api";
+import type TableType from "metabase-lib/metadata/Table";
+
+import { CollectionBadge } from "./CollectionBadge";
+import type { WrappedResult } from "./types";
+
+function getInfoText(result: WrappedResult) {
```

Comment:
Nit: not a huge deal but I think maybe we should keep this logic inside `InfoText` since we are using JSX here.

It feels a bit odd to return JSX in a helper method then have a wrapper component just to call the helper and return its JSX, as opposed to just defining it in the component.